### ISSUE TITLE
WebRTC not working

### ISSUE THREAD

**Issue by dpanic** [mattermost/desktop#1288]
I confirm (by marking "x" in the [ ] below: [x]):

- [ x ] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [ x ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [ x ] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
I am trying to embed JITSI Video in popup window, by using this url:

https://app-chat.[REDACTED].com/plugins/jitsi/test333

**Environment**
- Operating System: Linux Ubuntu 18.04.4
- Mattermost Desktop App version: 4.4.1
- Mattermost Server version: 5.22.1

**Steps to reproduce**
Just click link I provided.

**Expected behavior**
Expected microphone to work, camera to work. Be able to share screen.  Be able to send and receive WebRTC data. 

P.S. I tried to download your source code, upgraded Electron to latest upgreadable version, recompiled. It was running Chrome 81, same error.

**Observed behavior**
When I try to access Microphone/Camera this is error I get. No data in/out, just websocket is working. I was thinking that ORIGIN POLICY is in place, so I managed to put everything behind transparent reverse proxy my end, to make it under same ORIGIN, but no luck.

s {gum: {…}, name: "gum.general", message: "Failed to execute 'getUserMedia' on 'MediaDevices': Malformed constraints object.", stack: "Error↵    at new s

![Screenshot from 2020-05-08 22-55-37](https://user-images.githubusercontent.com/346557/81449312-4c4a4a80-9180-11ea-9a2a-b0bcfd1b189f.png)

**Comment by Willyfrog** [comment]
Hello! just to clarify: are you using jitsi as the server url or is this a url pasted in a link?

**Comment by dpanic** [comment]
> Hello! just to clarify: are you using jitsi as the server url or is this a url pasted in a link?

**Tried both options.** 

As I already mentioned, I did moved everything to same ORIGIN, same host, port protocol...Problem was in permissions. Audio is received. But nothing being sent over.

I did however managed to run once microphone and video by starting desktop app with modified code like this:

![Screenshot from 2020-05-12 11-04-39](https://user-images.githubusercontent.com/346557/81664998-703bb380-9440-11ea-82f9-db3972263d49.png)

**Comment by dpanic** [comment]
![Screenshot from 2020-05-12 11-06-42](https://user-images.githubusercontent.com/346557/81665198-ba249980-9440-11ea-988c-d5881d9fd811.png)

This one was working as I remember.

**Comment by Willyfrog** [comment]
not sure about linux, but for some operating systems you need to check for permissions before using media: https://www.electronjs.org/docs/api/system-preferences#systempreferencesaskformediaaccessmediatype-macos

**Comment by dpanic** [comment]
Ok about MacOS, but I am testing and developing on Linux.
My collegues do test on MacOS.

**Comment by Willyfrog** [comment]
the problem with enabling `nodeIntegration` is that we are running remote code from the server and from plugins, and it is considered insecure: https://www.electronjs.org/docs/tutorial/security#2-do-not-enable-nodejs-integration-for-remote-content

on the other hand, there are jitsi plugins: https://integrations.mattermost.com/?s=Jitsi&submit= although I don't think they embed conferencing inside of mattermost

**Comment by dpanic** [comment]
My idea is to make this work. And I will create PR for following:

/ext/insecure/jitsiURL

Something like that...
What do you think?

**Comment by Willyfrog** [comment]
sorry I'm not sure i understand: what does creating a PR for that path means? you want to enable nodeintegration for that path?

**Comment by dpanic** [comment]
Idea is to implement loading external JITSI url in a popup (modal window) with audio/video permissions.

**Comment by Willyfrog** [comment]
popups are allowed to run plugin's code, maybe you should check that. Also you might find some help in the [toolkit channel](https://community.mattermost.com/core/channels/developer-toolkit) on community server

**Comment by dpanic** [comment]
> popups are allowed to run plugin's code, maybe you should check that. Also you might find some help in the [toolkit channel](https://community.mattermost.com/core/channels/developer-toolkit) on community server

I did started plugin's code by issuing popup link with /plugins prefix. But that didn't work

**Comment by Willyfrog** [comment]
using a jitsi room as a server seems to work, so I'm guessing we are most likely blocking some request. Can you open the devtools and see if there is a request that it is either failing or not being answered?

**Comment by dpanic** [comment]
> using a jitsi room as a server seems to work, so I'm guessing we are most likely blocking some request. Can you open the devtools and see if there is a request that it is either failing or not being answered?

**using a jitsi room as a server seems to work** what do you mean by this?
I don't see any requests blocked. 

![Screenshot from 2020-05-26 21-53-40](https://user-images.githubusercontent.com/346557/82944510-a1b59280-9f9b-11ea-9cfb-6c8274843c60.png)

**Comment by dpanic** [comment]
I managed to move little bit forward.
Problem is in permissions.

File src/main.js
 L787: session.defaultSession.setPermissionRequestHandler((webContents, permission, callback) => {

**This must be modified to return true for popup window.**

**Comment by Willyfrog** [comment]
> using a jitsi room as a server seems to work what do you mean by this?
Sorry, let me rephrase: i meant going to the add new server, set a name and instead of a mattermost server url, add that of a previously created room from jitsi.
Jitsi room loads and cam and mic can be used.

> This must be modified to return true for popup window.

Would you be open to submit a PR for it?

**Comment by dpanic** [comment]
Yes definitelly. But I will need some guidelines from your side.

**Comment by Willyfrog** [comment]
sure! let me know how I can help you. if needed you can reach me on the [community server](https://community.mattermost.com/)
